docs/cloudevents/cloudevents-spans.md

```diff
@@ -35,168 +31,26 @@ consult the
-
-- What components in a system reacted to an event
-- What further events were sent due to an incoming event
-- Which event caused the exception
-
-With the conventions in this document, the application scenario above would
-produce a trace where it is possible to correlate a producer with a consumer(s):
-
-```
-+-------------------------------------------------------+
-|          Trace 1                                      |
-|                                                       |
-|          +---------------------------------------+    |
-|    +---> | Create event                          |    |
-|    |     +---------------------------------------+    |
-|    |     +---------------------------------------+    |
-|    |     | Send (auto-instr)                     |    |
-|    |     +---------------------------------------+    |
-|    |        +------------------------------------+    |
-|    |        | Intermediary: Received (auto-instr)|    |
-|    |        +------------------------------------+    |
-|    |        +------------------------------------+    |
-|    |        | Intermediary: Send (auto-instr)    |    |
-|    |Link    +------------------------------------+    |
-|    |                                                  |
-|    |                                                  |
-|    |                                                  |
-|    |     Trace 2                                      |
-|    |                                                  |
-|    |     +---------------------------------------+    |
-|    |     | Consumer: Receive (auto-instr)        |    |
-|    |     +---------------------------------------+    |
-|    |       +-------------------------------------+    |
-|    +------ | Consumer: Process                   |    |
-|            +-------------------------------------+    |
-|                                                       |
-+-------------------------------------------------------+
-```
-
 ## Conventions
 
-To achieve the trace above, it is necessary to capture the context of
-the event creation so that when the CloudEvent reaches its destination(s), this
-context can be continued. Each CloudEvent acts then as the medium of this
-context propagation.
-
-This document relies on the CloudEvents specification, which defines this
-context propagation mechanism via the
-[CloudEvents Distributed Tracing Extension](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/extensions/distributed-tracing.md).
-Once the trace context is set on the event
-via the Distributed Tracing Extension, it MUST not be modified.
-
-The remainder of this section describes the semantic conventions for Spans
-required to achieve the proposed trace.
-
-### Spans
+CloudEvents can be sent via various communication protocols. When the protocol
+used to send CloudEvents has a well established support for trace context propagation,
+instrumentations SHOULD propagate the context using this built-in mechanism.
```

Comment:
We shouldn't say that, it's misleading.

Given CloudEvents are sent via HTTP, then the established context propagation mechanism for HTTP doesn't satisfy [requirements for messaging context propagation](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/messaging-spans.md#context-propagation):

> If possible, the message creation context SHOULD be attached in such a way that it cannot be changed by intermediaries.

A context attached to a HTTP request by the producer will not make it through to the consumer. I'd remove this remark.

Reply:
I agree, but I'm not sure about removing it completely. It is perfectly fine to attach the context to the HTTP request, if the communication is done via HTTP and not via messaging systems. Previously I had paragraphs saying explicitly that if sending HTTP in per-to-per/HTTP the HTTP conventions/context propagation should be used, otherwise the messaging conventions are to be used. But I tried to incorporate Liudmilla's suggestions https://github.com/open-telemetry/semantic-conventions/pull/1182#discussion_r1657656417. 

The thing is, I'm not convinced that we should just talk about messaging here, as I'm sure people also use CloudEvents in synchronous HTTP use cases.

Reply:
Maybe we should have clear sections on this document to give directions on specific protocols/methods of sending CloudEvents? If in synchronous/peer-to-peer/HTTP -> use HTTP conventions and propagate via header. If using asynchronous/messaging -> Use Messaging conventions, propagate via existing messaging system mechanisms otherwise use trace extension from CloudEvents.

Reply:
I'd prefer to turn this around a bit. Instead of taking the usage of CloudEvents a starting point, I'd rather like to treat this as an implementation detail. Instrumentation authors should follow the following train of thought:

1. I'm doing peer-to-peer HTTP, or I'm doing messaging, or I'm doing something else.
2. Therefore, I'm instrumenting according to HTTP semantic conventions, to messaging semantic conventions, or according to another semantic conventions area (RPC, or whatever).
3. I'm using CloudEvents as data format.
4. Therefore, I'm adding CloudEvents attributes to the spans I obtained from following (2).

> Maybe we should have clear sections in this document to give directions on specific protocols/methods of sending CloudEvents?

I'd like to avoid that, as we can't cover all cases. What about CloudEvents via gRPC? AMQP? Stored in databases?

Reply:
But if they follow the train of thought you laid out above, wouldn't make sense that we have sections for each mechanism? 🤔 . For HTTP and Messaging we have it all covered, and for the rest, it seems fit to rely on the distributed tracing extension from cloud events as the propagation mechanism.